test/ecto/changeset_test.exs

```diff
@@ -1061,6 +1061,21 @@ defmodule Ecto.ChangesetTest do
     assert changeset.errors == [topics: {"can't be blank", [validation: :required]}]
   end
 
+  test "filter_required/2" do
+    # When valid
+    required =
+      changeset(%{"body" => "something"})
+      |> filter_required([:title, :body])
+
+    assert required == [:body]
+
+    # When unknown field
+    assert_raise ArgumentError, ~r/unknown field :bad in/, fn  ->
+      changeset(%{"title" => "hello", "body" => "something"})
+      |> filter_required(:bad)
+    end
+  end
+
```

Comment:
could you add a test like this: `changeset(%Foo{x: 1}, %{y: 2}); …; filter_required(changeset, [:x, :y])`, in other words, we look at both changeset.data and changeset.changes, right?

Reply:
thanks, done!